DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/PerturbProcessor.cs

```diff
@@ -0,0 +1,136 @@
+            { DicomVR.SL, new VRTypes() { ElementType = typeof(DicomSignedLong), ValueType = typeof(int[]) } },
+            { DicomVR.SS, new VRTypes() { ElementType = typeof(DicomSignedShort), ValueType = typeof(short[]) } },
+            { DicomVR.US, new VRTypes() { ElementType = typeof(DicomUnsignedShort), ValueType = typeof(ushort[]) } },
+            { DicomVR.OW, new VRTypes() { ElementType = typeof(DicomOtherWord), ValueType = typeof(ushort[]) } },
+            { DicomVR.UL, new VRTypes() { ElementType = typeof(DicomUnsignedLong), ValueType = typeof(uint[]) } },
+            { DicomVR.OL, new VRTypes() { ElementType = typeof(DicomOtherLong), ValueType = typeof(uint[]) } },
+            { DicomVR.UV, new VRTypes() { ElementType = typeof(DicomUnsignedVeryLong), ValueType = typeof(ulong[]) } },
+            { DicomVR.OV, new VRTypes() { ElementType = typeof(DicomOtherVeryLong), ValueType = typeof(ulong[]) } },
+            { DicomVR.SV, new VRTypes() { ElementType = typeof(DicomSignedVeryLong), ValueType = typeof(long[]) } },
+        };
+
+        public PerturbProcessor(JObject settingObject, IAnonymizerSettingsFactory settingFactory = null)
+        {
+            EnsureArg.IsNotNull(settingObject, nameof(settingObject));
+
+            settingFactory ??= new AnonymizerSettingsFactory();
+            var perturbSetting = settingFactory.CreateAnonymizerSetting<PerturbSetting>(settingObject);
+            _perturbFunction = new PerturbFunction(perturbSetting);
+        }
+
+        public void Process(DicomDataset dicomDataset, DicomItem item, ProcessContext context = null)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            if (item.ValueRepresentation == DicomVR.AS)
+            {
+                var values = ((DicomAgeString)item).Get<string[]>().Select(Utility.ParseAge).Select(x => _perturbFunction.Perturb(x));
+                dicomDataset.AddOrUpdate(item.ValueRepresentation, item.Tag, values.Select(Utility.AgeToString).ToArray());
+            }
+            else
+            {
+                if (!_numericValueTypeMapping.ContainsKey(item.ValueRepresentation))
+                {
+                    throw new AnonymizationOperationException(DicomAnonymizationErrorCode.UnsupportedAnonymizationFunction, $"Perturb is not supported for {item.ValueRepresentation}");
+                }
+
+                var elementType = _numericValueTypeMapping[item.ValueRepresentation].ElementType;
+                var valueType = _numericValueTypeMapping[item.ValueRepresentation].ValueType;
+
+                // Get numeric value using reflection.
+                var valueObj = elementType.GetMethod("Get").MakeGenericMethod(valueType).Invoke(item, new object[] { -1 });
+                PerturbNumericValue(dicomDataset, item, valueObj as Array);
```

Comment:
as keyword will result in null value if cast fails. We may need to check values != null in line80